sample-apps/ruby-rails-sample-app/app/controllers/application_controller.rb

```diff
@@ -0,0 +1,78 @@
+require 'net/http'
+
+def get_xray_trace_id(otel_trace_id_hex)
+    xray_trace_id = "1-#{otel_trace_id_hex[0..7]}-#{otel_trace_id_hex[8..otel_trace_id_hex.length]}"
+    { traceId: xray_trace_id }
+end
+
+class ApplicationController < ActionController::Base
+    def aws_sdk_call
+        @@tracer.in_span("get-aws-s3-buckets") do |span|
+            s3 = Aws::S3::Client.new
+            s3.list_buckets
+        end
+        render json: get_xray_trace_id(OpenTelemetry::Trace.current_span.context.hex_trace_id)
+    end
+    
+    def outgoing_http_call
+
+        @@tracer.in_span("outgoing-http-call") do |span|
+            uri = URI.parse("https://aws.amazon.com/")
+            http = Net::HTTP.new(uri.host, uri.port)
+            http.use_ssl = true
+
+            request = Net::HTTP::Get.new(uri.request_uri)
+            res = http.request(request)
+            span.set_attribute("signal", "trace")
+            span.set_attribute("language", "ruby")
+            span.set_attribute("statusCode", res.code)
+        end
```

Comment:
nit: can this piece be reused since it is used by here and in the `outgoing_sampleapp` method in case `count == 0`?

I meant, can we extract this as a method that can be reused?

Reply:
Yes that's true. I extracted the method out and made it a return a value 'res' to still be able to set the res.code in the attributes.